File: youlexuan-sellergoods-interface/src/main/java/com/offcn/sellergoods/service/SpecificationService.java
package com.offcn.sellergoods.service;

import com.offcn.entity.PageResult;
import com.offcn.entity.SpecificationVo;
import com.offcn.pojo.TbSpecification;

import java.util.List;

/*
规格服务层接口
 */
public interface SpecificationService {

    /*
    全部规格列表
     */
    public List<TbSpecification> selectAll();

    /*
    分页列表
     */
    public PageResult selectPage(int pageNum, int pageSize);

    /*
    增加规格
     */
    public void insert(SpecificationVo SpecificationVo);

    /*
    修改规格
     */
    public void update(SpecificationVo SpecificationVo);

    /*
    根据ID获取实体
     */
    public SpecificationVo selectOne(Long id);

    /*
    删除规格
     */
    public void delete(Long[] ids);

    /*
    条件查询
     */
    public PageResult findPage(TbSpecification tbSpecification, int pageNum, int pageSize);
}

File: youlexuan-manager-web/src/main/webapp/js/service/specificationService.js
/*
定义服务
*/
app.service('specificationService', function ($http) {

    //获取所有的规格
    this.selectAll = function () {
        return $http.get('../specification/selectAll.do');
    };

    //分页
    this.selectPage = function (page, rows) {
        return $http.get('../specification/selectPage.do?page=' + page + '&rows=' + rows);
    };

    //增加规格
    this.insert = function (entity) {
        return $http.post('../specification/insert.do', entity);
    };

    //更新规格
    this.update = function (entity) {
        return $http.post('../specification/update.do', entity);
    };

    //查询单个
    this.selectOne = function (id) {
        return $http.get('../specification/selectOne.do?id=' + id);
    };

    //删除规格
    this.delete = function (ids) {
        return $http.get('../specification/delete.do?ids=' + ids);
    };

    //条件查询
    this.search = function (page, rows, searchEntity) {
        return $http.post('../specification/search.do?page=' + page + "&rows=" + rows, searchEntity);
    };

})